dashboard/app/models/user.rb

```diff
@@ -2261,6 +2261,41 @@ def providers
     end
   end
 
+  # Returns number of times a user has attempted to join a section in the last 24 hours
+  # Returns nil if no section join attempts
+  def num_section_attempts
+    properties['section_attempts'] || 0
+  end
+
+  # There aare two possible states in which we would want to reset section attempts
+  # 1) Initialize for the first time 2) 24 hours have passed since last attempt
+  def reset_section_attempts?
+    !num_section_attempts || !properties['section_attempts_last_reset'] || num_section_attempts == 0 || (DateTime.now - DateTime.parse(properties['section_attempts_last_reset'])).to_i > 0
```

Comment:
does `DateTime`arithmetic operate on "day units"? to me, `(DateTime.now - DateTime.parse(properties['section_attempts_last_reset'])).to_i > 0` looks like you're checking to see if any time has passed at all, so if that difference is operating on day units, could you add that to the comment? if not, i think you want something other than 0 here (depending on the units)

Reply:
Subtractiing DateTime objects in Ruby gives the difference in days as a floating point number: https://stackoverflow.com/questions/30953974/datetime-subtraction-in-ruby-2

The key here is the `to_i` casting. If the two objects are less than 1 day apart, the subtraction will yield a fractional number < 1. By casting to an int, we can check to ensure at least 1 full day has passed. Does this make sense? I'll work to clarify with an inline comment.